### ISSUE TITLE
3.2.4b2 - Vertex Shaders behave unexpected with Batching (GLES3) enabled.

### ISSUE THREAD

**Issue by RPicster** [godotengine/godot#43673]
**Godot version:**
3.2.4 beta 2 - official build (Win 64)

**OS/device including version:**
Windows 10 Pro - 10.0.19042
Nvidia RTX 2080

**Issue description:**
With batching enabled in GLES3, rendering of the vertex function of a shader creates unexpected results, especially in combination with viewports

Without batching:
![image](https://user-images.githubusercontent.com/9423774/99642126-6cba6200-2a4b-11eb-94f3-c0e3e8b8479a.png)

With batching:
![image](https://user-images.githubusercontent.com/9423774/99642261-970c1f80-2a4b-11eb-80eb-01a84641804d.png)

This is especially destructive when using the vertex shader on a ViewportContainer, the content does not get rendered at all (in my reprex - "viewport_variant.tscn" the viewports content is not rendered at all; In my game, the contents are completely weird)

**Steps to reproduce:**
* Create a new 2D Scene
* Add a Sprite Node
* Give it some texture (icon.png)
* Add a Shader Material
* Add a Shader with the following shader code:

```
shader_type canvas_item;

void vertex() {
	VERTEX += vec2(0.0,0.0);
}
```

It also works with the following code, but the above code gives some extra strange behaviour:

```
shader_type canvas_item;

void vertex() {
	VERTEX = VERTEX;
}
```

**Minimal reproduction project:**
[3.2.4b2_shader_problem.zip](https://github.com/godotengine/godot/files/5565505/3.2.4b2_shader_problem.zip)

**Comment by lawnjelly** [comment]
Ah yes looks like I'm not writing the colors in this situation for the main scene (is large vertex format, for batching custom shaders). This should be easy to fix, I'm amazed it has not come up before.

**Comment by akien-mga** [comment]
Fixed by #43679.
